libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt

```diff
@@ -24,9 +26,11 @@ interface SecureChannel : ProtocolBinding<SecureChannel.Session> {
          */
         val remotePubKey: PubKey,
 
-        /** The id of the next protocol, used to select the muxer
-         *
+        /**
+         * Contains muxer if security protocol supports
+         * [Early Multiplexer Negotiation](https://docs.libp2p.io/concepts/multiplex/early-negotiation/)
+         * and the protocol was successfully negotiated. Else contains `null`
          */
-        val nextProto: String
+        val earlyMuxer: NegotiatedStreamMuxer?
```

Comment:
Is it worth making this an Optional?

Reply:
It's kind of 'optional' in the sense of 'nullable'

Reply:
Yep, I just meant using the type system to make it obvious.

Reply:
Maybe this is the Kotlin way to do that?